test/integration/builder/selects.js

```diff
@@ -178,6 +179,21 @@ module.exports = function(knex) {
       });
     });
 
+    it('emits error on the stream, if not passed a function, and connecting fails', function(done) {
+      var expected = new Error();
+      var original = Runner.prototype.ensureConnection;
+      Runner.prototype.ensureConnection = function() {
+        return Promise.reject(expected);
+      };
+      var stream = knex('accounts').stream();
+      stream.on('error', function(actual) {
```

Comment:
maybe this could be wrapped to 

```
return new Promise((resolve, reject) => { ... });
```

which will reject after 5ms or so unless error was emitted? That way test won't hang until mocha timeout and cleaning up could also restore the `Runner.prototype.ensureConnection = original;` even when test fails?